woocommerce/woocommerce-android
# This commit adds order_key metadata into the PaymentInfo.kt

Parent issue: #4030 


### Description
This PR adds order_key metadata into the PaymentInfo. [Here's ](https://github.com/woocommerce/woocommerce-ios/issues/4077)some more info on this. [This](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2105) is the corresponding FluxC PR for this.

### Note
Please make sure [corresponding fluxc](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2105) PR is merged before merging this one.


### Testing instructions
1. Add Log or debug pointer at line no 81 [map[MetaDataKeys.ORDER_KEY.key]]
2. Run the app and click on the `orders` tab
3. Click on any of the order which has `Processing` status
4. Click on the `Collect payment` button in the order detail screen
5. Verify from the log or debug pointer that you are getting order_key field with some value that is not null or empty
6. Run unit tests and ensure it passes
